packages/ruby/lib/readme/mask.rb

```diff
@@ -0,0 +1,11 @@
+require 'digest'
+
+module Readme
+    class Mask
+        def self.mask(data)
+            digest = Digest::SHA2.new(512).base64digest(data)
+            opts = data.length >= 4 ? data[-4,4] : data
```

Comment:
Ruby doesn't have a clean method to get the last `n` characters from a string. The closest is `data[-4,4]`, BUT will return `nil` if the string has less than 4 characters. So dumb.